R/Rules-class.R

```diff
@@ -1104,57 +1104,61 @@ IncrementsMin <- function(increments_list) {
 #' @export
 #'
 setClass(
-  Class = "Stopping",
-  contains = list("VIRTUAL")
+  Class = "Stopping"
 )
 
+# StoppingCohortsNearDose ----
 
-## --------------------------------------------------
-## Stopping based on number of cohorts near to next best dose
-## --------------------------------------------------
-
-##' Stop based on number of cohorts near to next best dose
-##'
-##' @slot nCohorts number of required cohorts
-##' @slot percentage percentage (between 0 and 100) within the next best dose
-##' the cohorts must lie
-##'
-##' @example examples/Rules-class-StoppingCohortsNearDose.R
-##' @keywords classes
-##' @export
-.StoppingCohortsNearDose <-
-    setClass(Class="StoppingCohortsNearDose",
-             representation(nCohorts="integer",
-                            percentage="numeric"),
-             prototype(nCohorts=2L,
-                       percentage=50),
-             contains="Stopping",
-             validity=function(object){
-                 o <- Validate()
+## class ----
 
-                 o$check((object@nCohorts > 0L) && is.scalar(object@nCohorts),
-                         "nCohorts must be positive scalar")
-                 o$check(is.probability(object@percentage / 100),
-                         "percentage must be between 0 and 100")
+#' `StoppingCohortsNearDose`
+#'
+#' @description `r lifecycle::badge("stable")`
+#'
+#' [`StoppingCohortsNearDose`] is the class for stopping based on number of
+#' cohorts near to next best dose.
+#'
+#' @slot nCohorts (`number`)\cr number of required cohorts.
+#' @slot percentage (`number`)\cr percentage (between 0 and 100) within the
+#'   next best dose the cohorts must lie.
+#'
+#' @aliases StoppingCohortsNearDose
+#' @export
+#'
+.StoppingCohortsNearDose <- setClass(
+  Class = "StoppingCohortsNearDose",
+  slots = c(
+    nCohorts = "integer",
+    percentage = "numeric"
+  ),
+  prototype = prototype(
+    nCohorts = 2L,
+    percentage = 50
+  ),
+  contains = "Stopping",
+  validity = v_stopping_cohort_near_dose
```

Comment:
```suggestion
  validity = v_stopping_cohorts_near_dose
```
to be consistent with class name